### PR TITLE
feat(ng-deploy): add option to specify firebaseProject (#2281) (#2063)

### PR DESCRIPTION
**Pull request by george43g**
### Checklist

   - Issue number for this PR: #2063 , #2281 
   - In a clean directory, `yarn install`, `yarn test` run successfully? yes

### Description

As per #2063 , #2281, I have added the option for the target firebase project to be specified in angular.json. Previously, the builder would simply guess the target project based on the first target which contained the same name/host.

### Code sample

In angular.json, you can now specify:
```json
"deploy": {
          "builder": "@angular/fire:deploy",
          "options": {
            "firebaseProject": "projectNameOrAlias" // <-- This is the new option.
          }
}
```
When running `ng deploy`, the CLI will confirm which firebase project is being deployed to. This value is optional - if not supplied, builder will fall back to old behaviour of guessing. If the project value is invalid, it will fail gracefully with a descriptive error message.

This feature is incomplete because `configurations` are not yet working, however, this seems to be an issue with ng-cli. IE there is no way to have multiple configurations (production, dev, staging, etc..) yet.

### CODE REVIEW

**Comment by Splaktar** [comment]
> there is no way to have multiple configurations (production, dev, staging, etc..) yet

This isn't the case. See [this angular.json config](https://github.com/angular-hispano/angular-hispano/blob/d68c47da2de5a26bbff6c90b3fecb92034219608/angular.json#L43-L102) and [these npm scripts](https://github.com/angular-hispano/angular-hispano/blob/d68c47da2de5a26bbff6c90b3fecb92034219608/package.json#L11-L14).

**Comment by george43g** [comment]
> > there is no way to have multiple configurations (production, dev, staging, etc..) yet
> 
> This isn't the case. See [this angular.json config](https://github.com/angular-hispano/angular-hispano/blob/d68c47da2de5a26bbff6c90b3fecb92034219608/angular.json#L43-L102) and [these npm scripts](https://github.com/angular-hispano/angular-hispano/blob/d68c47da2de5a26bbff6c90b3fecb92034219608/package.json#L11-L14).

I should probably update some maintainers about this. So although it's possible using npm scripts, it wasn't possible using just a angular.json configuration. I've already submitted a request to fix the bug in the angular CLI and there's a pending merge: https://github.com/angular/angular-cli/issues/17332

There are some `ng deploy` builders that actually rely on a `--configuration` flag to determine the *build target*, which is a mistake. Those setups will be broken by this update, I'll give them a heads up.

**Comment by Splaktar** [comment]
Ah, thank you for the clarification and bug report! Glad to see that Alan already has a PR to fix this.

**Comment by george43g** [comment]
Hey, sorry I'm a bit new to this so maybe you can explain how it all works...
I made a couple of commits a few weeks back where I updated the docs and added a few small features.

So, it says that travisCI is still pending, but when I click on it it says it passed all tests... is this stopping the commits from being merged?

Have they been merged? It seems like it's still waiting for approval. Does it usually take a while?
And finally - it's also saying the branch is out of date - is there any point in merging the latest changes?
Sorry I'm not 100% sure what to do but I was hoping to start using the new features I wrote soon :)

File: src/schematics/deploy/actions.ts [hunk 1]
```diff
@@ -159,7 +159,9 @@ export default async function deploy(
   }
 
   try {
-    await firebaseTools.use(firebaseProject, { project: firebaseProject });
+    await firebaseTools.use(firebaseProject, { project: firebaseProject }).then(() => {
+      context.logger.info(`🔥 Your configured Firebase project is '${firebaseProject}'`);
```
Comment (NothingEverHappens) [inline src/schematics/deploy/actions.ts:163]: You don't need .then here,
just log on the next line.

File: src/schematics/deploy/schema.json [hunk 2]
```diff
@@ -9,6 +9,10 @@
       "description": "Target to build.",
       "pattern": "^[^:\\s]+:[^:\\s]+(:[^\\s]+)?$"
     },
+    "firebaseProject": {
+      "type": "string",
+      "description": "The firebase project name or project alias to use when deploying"
```
Comment (NothingEverHappens) [inline src/schematics/deploy/schema.json:14]: nit: Please capitalize "Firebase"

**Comment by Splaktar** [comment]
@george43g Travis says that the tests passed, but it seems like GitHub hasn't figured that out. If you do another push with some of the changes from @NothingEverHappens' suggestions, it should clear it up.

Your changes in this PR have not yet been merged. Getting changes merged can take weeks or months and requires a lot of patience.

The branch being out of date isn't an issue at this time. If it lists some conflicts, then you would need to merge in the latest changes and resolve the conflicts.

At this point, it's just waiting on @jamesdaniels or @davideast to do final approval and get it merged.

File: src/schematics/deploy/actions.ts (hunk 1 above)
Comment (george43g) [reply src/schematics/deploy/actions.ts]: Done!

File: src/schematics/deploy/schema.json (hunk 2 above)
Comment (george43g) [reply src/schematics/deploy/schema.json]: Also done! :)

**Comment by jamesdaniels** [comment]
I'm working to cut 6.0.1 and support upcoming NG 10 release right now. Our CI/CD pipeline got messed up which is causing some backlog :( Will cut 6.1 with some new features shortly thereafter and I'm planning on getting this in.

**Comment by george43g** [comment]
> I'm working to cut 6.0.1 and support upcoming NG 10 release right now. Our CI/CD pipeline got messed up which is causing some backlog :( Will cut 6.1 with some new features shortly thereafter and I'm planning on getting this in.

Any chance we can get this in now? I'm working for a company as a devops role atm and we could really use this feature :D

**Comment by jamesdaniels** [comment]
Solving conflicts and merging into #2647

**Pull request closed without merging**